example_queries/get_service_name.rs.ref

```diff
@@ -181,83 +254,76 @@ impl Filter {
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(",");
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(&prop_str);
-            }
-        }
-        else {
-            x.headers.insert("properties_service_name".to_string(), prop_str);
-        }
-         prop_str = format!("{whoami}.{property}=={value}",
-                                                      whoami=&self.whoami.as_ref().unwrap(),
-                                                      property="service_name",
-                                                      value=self.filter_state["node.metadata.WORKLOAD_NAME"].string_data.as_ref().unwrap().to_string());
-                                             
-        if x.headers.contains_key("properties_service_name") {
-            if !x.headers["properties_service_name"].contains(&prop_str) { // don't add our properties if they have already been added
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(",");
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(&prop_str);
+        // Fetch ferried data
+        let mut ferried_data: FerriedData;
+        if !x.headers.contains_key("ferried_data") {
+            ferried_data = FerriedData::default();
+        } else {
+            match serde_json::from_str(&x.headers["ferried_data"]) {
+                Ok(fd) => { ferried_data = fd; }
+                Err(e) => {
+                    log::error!("could not translate stored data to json string: {0}\n", e);
+                    return vec![x];
+                }
             }
         }
-        else {
-            x.headers.insert("properties_service_name".to_string(), prop_str);
-        }
-         
-        self.store_headers(x.clone());
-        return vec!(x);
+
+        // Insert properties to collect
+        let mut prop_tuple;
+        prop_tuple = Property::new(self.whoami.as_ref().unwrap().to_string(),
+                                                   "service_name".to_string(), 
+                                                   self.filter_state["node.metadata.WORKLOAD_NAME"].clone());
+                                             ferried_data.unassigned_properties.push(prop_tuple); prop_tuple = Property::new(self.whoami.as_ref().unwrap().to_string(),
+                                                   "service_name".to_string(), 
+                                                   self.filter_state["node.metadata.WORKLOAD_NAME"].clone());
+                                             ferried_data.unassigned_properties.push(prop_tuple); 
+
+        // Return ferried data to x, and store headers
+        put_ferried_data_in_hdrs(&mut ferried_data, &mut x.headers);
+        self.store_headers(x.uid, x.headers.clone());
```

Comment:
What is the plan to implement those on the filter? We do not have "x" there.

Reply:
All calls to x.uid would be replaced with the trace_id value you have at the beginning of all of the on_http_respones/request_inbound/outbound values.  So what gets put in trace_id from

if let Some(trace_id_) = self.get_http_request_header("x-request-id") {
    trace_id = trace_id_;
}

Reply:
x.headers.clone()

Would probably be a function that takes no arguments but self, and has a loop like

for (name, value) in &self.get_http_request_headers() {
 // do stuff
}

Reply:
Oh yeah you are right, I was thinking of the shared data. Let's go with all of this for now.